library/core/src/iter/traits/iterator.rs

```diff
@@ -285,6 +289,7 @@ pub trait Iterator {
     /// ```
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[rustc_do_not_const_check]
     fn last(self) -> Option<Self::Item>
```

Comment:
Can be removed simply by making the inner function const.
In fact looking at this most of those seem to be no longer needed now that `const_closures` is in bootstrap.